### ISSUE TITLE
Daily Sales Invoice report - Serial Number column

### ISSUE THREAD

**Issue by bkm888** [mntechnique/jmi_erpnext#72]
Can you confirm that this column has the necessary logic to populate itself if a serial number is actually found in the ERPNext data?

BKM

**Comment by ghost** [comment]
@bkm888 

As mentioned in the mail if the products that you are going to resale, are going to be the new products in the system, in that case, you while creating new items you can check 'has serial no.'

Also, We did observe that serial numbers are not being used in your system for existing items. 
As we cannot assign serial number/batch numbers to the 'Enabled' items,
How we can do it:
1. Disable the existing item
2. Duplicate the same item
3. This time tick 'has serial no.' and save

**Comment by bkm888** [comment]
@Siddhisawant8 

You are correct, the client is not using serial numbers at this time. They expect their first shipment of more expensive serial numbered machines in September. The product they are waiting for is being developed exclusively for them. To that end I attempted to add serial numbered items to the system in a slightly different way than you suggested because I need to know how to do this before September anyway.
 
Instead of using the steps to disable, duplicate, and set "has serail no." flag, I just created a new Item in the system. 

The trouble came when I tried to place a purchase order for the item and then a Purchase Receipt for the item so it could go into inventory the correct way. I could not find a way to receive the item. It kept demanding a serial number for the item but gave me no place to enter a serial number. 

This is why I am having trouble testing serial numbered items in the report. I am not able to receive one into inventory.

If you look in the GCP test server, you will find an Item with Item code MS900999 named a SUPER-DUPER POLISHER. I created it, generated a PO for it, and tried to generate a Purchase Receipt for it and that is where the process broke down. 

I was hoping to follow the correct path to get the item into inventory and then sell it so I could follow it onto the Daily Sales Invoice report.

At this point I am not sure how to proceed with the test.

BKM

**Comment by ghost** [comment]
@bkm888

I had a chance to go through your test server, for the new serialized item created MS900999.

1. This is the Item MS900999 created by you, which has 'has serial no' ticked and no 'Serial Number Series' assigned.
![itemnoseriesgiven](https://user-images.githubusercontent.com/29235673/41653219-00dea3da-74a3-11e8-942d-9257430f38fa.png)


In this case **while creating a Purchase Receipt form a Purchase Order, at Item level we need to enter the serial number manually.** 
![snumberitem](https://user-images.githubusercontent.com/29235673/41653906-3ce763d8-74a5-11e8-99b6-3918b7a978f4.gif)





2. This is the Item MS900999-1  created, which has 'has serial no' ticked and  'Serial Number Series' assigned manually.
![itemwithseriesgiven](https://user-images.githubusercontent.com/29235673/41653583-26164c92-74a4-11e8-940b-877cd95c27b4.png)


And here, **since the series is mentioned, serial numbers are auto-generated after submission of Purchase Receipt.**
![itemwithseriesgiven](https://user-images.githubusercontent.com/29235673/41653729-a22dd610-74a4-11e8-846a-b6a384c77ec4.png)

**Comment by ghost** [comment]
@bkm888

I went ahead and created a sales invoice of one of the serialized item, It does reflect in the report.

![dailyreport_serialnumber](https://user-images.githubusercontent.com/29235673/41654064-dabe78b2-74a5-11e8-916f-5017c84f18b2.png)
